src/duqtools/ids/_merge.py

```diff
@@ -29,37 +29,31 @@ def merge_data(
     variables : Sequence[IDSVariableModel]
         variables
     """
-
-    # make sure we do not mess up input arguments
-    variables = variables.copy()
-
-    # Sometimes, dimensions are not yet included,
-    # We can safely include them into the variables manually, as they are
-    # automatically transformed into coordinates, and therefore not written back
-    variable_dict = {variable.name: variable for variable in variables}
+    # Add dimensions to variables
+    variable_dict = {}
+    
     for variable in variables:
+        variable_dict[variable.name] = variable
+
         for dim in variable.dims:
-            # TODO time should also be $ prefixed, remove the `or time` when fixed
-            if not (dim.startswith('$') or dim.startswith('time')):
-                continue
-            name = dim.lstrip('$')
-            if name not in variable_dict:
-                variables.append(var_lookup[name])
-                variable_dict[name] = variables[-1]
+            dim_var = var_lookup[name]
+            variable_dict.setdefault(dim_var.name, dim_var)
+
+    variables = tuple(variable_dict.keys())
```

Comment:
1. `keys()` is default
2. should be `values()`
```suggestion
    variables = tuple(variable_dict.values())
```